apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx

```diff
@@ -51,13 +60,34 @@ export const SettingsProfileKeysPage = () => {
   }
   const closeSshKeyDialog = () => setSshKeyDialog(false)
 
+  const closeAlertDeleteDialog = () => setIsAlertDeleteDialogOpen(false)
+  const openAlertDeleteDialog = ({ identifier, type }: AlertDeleteParams) => {
+    setIsAlertDeleteDialogOpen(true)
+    setAlertParams({ identifier, type })
+  }
+
   const queryParams: ListPublicKeyQueryQueryParams = {
     page: 1,
     limit: 30,
     sort: 'created',
     order: 'asc'
   }
 
+  useListTokensQuery(
```

Comment:
why are we maintaining state ourselves? can we simply use the state reference returned by the hook?

```
const { data, error } = useListTokensQuery()
```

Reply:
nevermind.. realised we need this for optimistic updates